File: Examples/ObjcDemo/RixSDKPodDemo/RixEngineObjcDemo/UI/UnityLevelPlay/LevelPlayMainVC.m
#import "LevelPlayMainVC.h"
#import "LevelPlayBannerVC.h"
#import "LevelPlayRewardVideoVC.h"
#import "LevelPlayInterstitialVC.h"
#import <IronSource/IronSource.h>
#import "AdConfig.h"

@implementation LevelPlayMainVC

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = NSLocalizedString(@"levelPlay_ad", @"");
}

- (MenuAppearance)menuAppearance {
    return MenuAppearanceCard;
}

- (NSArray<MenuItem *> *)menuItems {
    return @[
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"banner_ad", @"") makeVC:^UIViewController *{ return [[LevelPlayBannerVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"rewardVideo_ad", @"") makeVC:^UIViewController *{ return [[LevelPlayRewardVideoVC alloc] init]; }],
        [[MenuItem alloc] initWithTitle:NSLocalizedString(@"interstitial_ad", @"") makeVC:^UIViewController *{ return [[LevelPlayInterstitialVC alloc] init]; }]
    ];
}

- (NSString *)menuSubtitleAtIndex:(NSInteger)index {
    NSArray *subtitles = @[
        @"Flexible formats at the top, middle or bottom of your app.",
        @"Users engage with a video ad in exchange for in-app rewards.",
        @"Full-screen ads at natural breaks or transition points."
    ];
    if (index >= 0 && index < subtitles.count) {
        return subtitles[index];
    }
    return nil;
}

- (void)setupSDK {
    LPMInitRequest *initRequest = [[[LPMInitRequestBuilder alloc] initWithAppKey: AdConfig.LevelPlay_App_Key] build];
    [LevelPlay initWithRequest:initRequest completion:^(LPMConfiguration * _Nullable config, NSError * _Nullable error) {
        if (error) {
            NSLog(@"LevelPlayMainVC: initSDK failed: %@", error.localizedDescription);
        } else {
            NSLog(@"LevelPlayMainVC: initSDK success");
        }
    }];
}

@end

File: SwiftPodDemo/AlxAdsDemo/UI/UnityLevelPlay/LevelPlayMainVC.swift
//
//  LevelPlayMainVC.swift
//  AlxAdsDemo
//
//  Created by YXk on 2026/4/1.
//

import UIKit
import IronSource

// MARK: - Unity LevelPlay Ad Test | Main Page
class LevelPlayMainVC: BaseMenuVC {
    override var menuItems: [MenuItem] {[
        (title: NSLocalizedString("banner_ad",       comment: ""), makeVC: { LevelPlayBannerVC() }),
        (title: NSLocalizedString("rewardVideo_ad",  comment: ""), makeVC: { LevelPlayRewardVideoVC() }),
        (title: NSLocalizedString("interstitial_ad", comment: ""), makeVC: { LevelPlayInterstitialVC() }),
    ]}

    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = NSLocalizedString("levelPlay_ad", comment: "")
    }

    override var menuAppearance: MenuAppearance { .card }

    override func menuSubtitle(at index: Int) -> String? {
        let subtitles = [
            "Flexible formats at the top, middle or bottom of your app.",
            "Users engage with a video ad in exchange for in-app rewards.",
            "Full-screen ads at natural breaks or transition points."
        ]
        guard index >= 0, index < subtitles.count else { return nil }
        return subtitles[index]
    }

    override func setupSDK() {
        let initRequest = LPMInitRequestBuilder(appKey: AdConfig.LevelPlay_App_Key).build()
        LevelPlay.initWith(initRequest) { _, error in
            if let error {
                NSLog("LevelPlayMainVC: initSDK failed: %@", error.localizedDescription)
            } else {
                NSLog("LevelPlayMainVC: initSDK success")
            }
        }
    }
}
